Source/inv.cpp

```diff
@@ -1362,6 +1362,71 @@ bool AutoPlaceItemInInventory(Player &player, const Item &item, bool persistItem
 	app_fatal(StrCat("Unknown item size: ", itemSize.width, "x", itemSize.height));
 }
 
+std::vector<int> SortItemsBySize(Player &player)
+{
+	std::vector<std::pair<Size, int>> itemSizes; // Pair of item size and its index in InvList

```

Comment:
i guess you could `reserver(_pNumInv)`, a static array of `InventoryGridCells` could also work since 320bytes isn't exactly huge.